src/core/featuremodel.cpp

```diff
@@ -80,7 +80,7 @@ void FeatureModel::setFeatures( const QList<QgsFeature> &features )
         const int attributeIndex;
         const QgsFeature &feature;
         AttributeNotEqual( const QgsFeature &feature, int attributeIndex ) : attributeIndex( attributeIndex ), feature( feature ) {}
-        bool operator()( const QgsFeature &f) const { return feature.attributes().at( attributeIndex ) != f.attributes().at( attributeIndex ); }
+        bool operator()( const QgsFeature &f) const { return f.attributes().size() > attributeIndex && feature.attributes().at( attributeIndex ) != f.attributes().at( attributeIndex ); }
```

Comment:
Did you get a crash here? Isn't that caught already on line 75?

Reply:
I was getting a crash indeed, which this fixed.